trace/api/service/instructor_service.py

```diff
@@ -1,10 +1,17 @@
 from sqlalchemy_utils import sort_query
 
+from api.model.course import Course
 from api.model.instructor import Instructor
 
 
-def get_all_instructors(page, page_size, sort):
-    sql_results = sort_query(Instructor.query, sort).paginate(page, page_size, False).items
+def get_all_instructors(page, page_size, sort, term_id=None, department_id=None):
+    query = Instructor.query
+    if term_id is not None or department_id is not None:
+        filter_params = {**(term_id is not None and {'term_id': term_id} or {}),
```

Comment:
Python magic:

if term_id -> add the term_id filter to filter_params
if department_id -> add department_id filter to filter_params

But it's all nice and one line